opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -736,8 +745,23 @@ func generateVm(d *schema.ResourceData) (string, error) {
 	log.Printf("Number of CONTEXT vars: %d", len(context))
 	log.Printf("CONTEXT Map: %s", context)
 
-	for key, value := range context {
-		tpl.AddCtx(vmk.Context(key), fmt.Sprint(value))
+	if tplContext != nil {
+
+		// update existing context and add it to the current VM template
+		for key, value := range context {
+			keyUp := strings.ToUpper(key)
+			tplContext.Del(keyUp)
```

Comment:
maybe add a comment saying that if the KEY exists in Template and VM, the value from VM is kept.